fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreBase.java

```diff
@@ -1654,6 +1700,11 @@ default int getPipelineSize(@Nonnull PipelineOperation pipelineOperation) {
         return FDBQueriedRecord.indexed(indexedRecord);
     }
 
+    @Nonnull
+    default FDBQueriedRecord<Message> queriedRecordOfMessage(@Nonnull FDBIndexedRecord<Message> indexedRecord) {
```

Comment:
This suggests that the generic typing of the cursor isn't quite right.

Reply:
I couldn't make this work with the existing scheme, the issue being (AFAICT) with the qualifying of `<M extends Message>` in `FDBRecordStoreBase` as a `<Message>` in `FDBRecordStore`
(Meaning `FDBRecordStore implements FDBRecordStoreBase<Message>` makes the capture <M extends Message> in the record store implementation incompatible with the signature defined in FDBRecordStoreBase)
I would be happy to make this right, if you could offer guidance.

Reply:
> I would be happy to make this right, if you could offer guidance.

I outlined how I think it can work. I don't know if this is the same as what you tried before.

Should then be able to replace uses of this method with `queriedRecord`.

Reply:
Done. The trick was the missing serializer. Adding the serializer to the signature tied the <M extends Message> from the return type to the <M> in the method signatures. This is now behaving as expected and the redundant method was removed.